src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateNewObject = Object.assign({}, state);
```

Comment:
Consider renaming `stateNewObject` to `stateCopy` to improve code readability and adhere to naming conventions.